torch_xla/csrc/reduction.cpp

```diff
@@ -26,13 +27,11 @@ ReductionInfo GetReductionInfo(xla::XlaOp input, const xla::Shape& shape,
                                absl::Span<const xla::int64> dimensions,
                                bool keep_reduced_dimensions) {
   ReductionInfo rinfo;
-  size_t idim = 0;
+  std::unordered_set<xla::int64> reduced_dimensions(dimensions.begin(),
+                                                    dimensions.end());
   for (xla::int64 i = 0; i < shape.rank(); ++i) {
-    if (idim < dimensions.size() && dimensions[idim] == i) {
-      ++idim;
-      if (keep_reduced_dimensions) {
```

Comment:
hmmm do we no longer consider `keep_reduced_dimensions` before pushing to `rinfo.new_dimensions`?

Reply:
oh... missed this check..